python_modules/dagster/dagster/_core/ext/utils.py

```diff
@@ -205,9 +205,11 @@ def ext_protocol(
     ) as ci_params, message_reader.read_messages(
         msg_handler,
     ) as mr_params:
-        yield ExtOrchestrationContext(
+        ext_context = ExtOrchestrationContext(
             context_data=context_data,
             message_handler=msg_handler,
             context_injector_params=ci_params,
             message_reader_params=mr_params,
         )
+        yield ext_context
+        ext_context.is_task_finished = True
```

Comment:
may need `finally` block protection to ensure this is set on exception driven exit ? trying to think if theres a case where you exit CM via exception and still have access to ExtOrchestrationContext such that it may be called

Reply:
This needs a test either way to confirm behavior